src/ui/mod.rs

```diff
@@ -18,18 +18,14 @@ pub fn get_ascii_colors(
         dominant_language
     };
 
-    let colors = language.get_colors(true_color);
+    let mut colors: Vec<DynColors> = ascii_colors.iter().map(|s| num_to_color(s)).collect();
```

Comment:
```rust
let mut language_colors: Vec<DynColors> = language.get_colors(true_color);

if ascii_colors.is_empty() {
    language_colors
} else {
    let mut colors: Vec<DynColors> = ascii_colors.iter().map(|s| num_to_color(s)).collect();

    if language_colors.len() > colors.len() {
        let mut missing = language_colors.drain(colors.len()..).collect();
        colors.append(&mut missing);
    }
    colors
}
```
minor: for better performance --> exit early when `ascii_collors` is empty.

Reply:
Thanks for the suggestion, I've added it.